<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Envelope_Dlg</class>
 <widget class="QWidget" name="Envelope_Dlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1219</width>
    <height>911</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="title">
        <string>Mouse Configuration</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Zoom Axis:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBox_zoomx">
          <property name="text">
           <string>x</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBox_zoomy">
          <property name="text">
           <string>y</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Edit Axis:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBox_editx">
          <property name="text">
           <string>x</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QCheckBox" name="checkBox_edity">
          <property name="text">
           <string>y</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="title">
        <string>ControlPoints</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_New">
            <property name="text">
             <string>New</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="pushButton_Delete">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Delete</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_P">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Previous</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButton_N">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Next</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_9">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox_cpType">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>-1</number>
            </property>
            <item>
             <property name="text">
              <string>Step</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Linear</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>TCB</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>x:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_cpX">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string/>
            </property>
            <property name="maxLength">
             <number>32</number>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>y:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEdit_cpY">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="maxLength">
             <number>32</number>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_6">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Tension:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_cpT">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="minimum">
             <double>-1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Continuity:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_cpC">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="minimum">
             <double>-1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Bias:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_cpB">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="minimum">
             <double>-1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="m3d::EnvelopeGraph" name="graph" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="contextMenuPolicy">
      <enum>Qt::CustomContextMenu</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>m3d::EnvelopeGraph</class>
   <extends>QWidget</extends>
   <header>envelopegraph.h</header>
   <container>1</container>
   <slots>
    <signal>selectionChanged(unsigned)</signal>
    <signal>addControlPoint(value,value)</signal>
    <signal>moveControlPoint(value,value)</signal>
    <signal>deleteControlPoint()</signal>
    <slot>enableZoomX(bool)</slot>
    <slot>enableZoomY(bool)</slot>
    <slot>enableEditX(bool)</slot>
    <slot>enableEditY(bool)</slot>
    <slot>nextPoint()</slot>
    <slot>prevPoint()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>checkBox_zoomx</sender>
   <signal>toggled(bool)</signal>
   <receiver>graph</receiver>
   <slot>enableZoomX(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_zoomy</sender>
   <signal>toggled(bool)</signal>
   <receiver>graph</receiver>
   <slot>enableZoomY(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph</sender>
   <signal>selectionChanged(unsigned)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>updateControlPointUI(unsigned)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph</sender>
   <signal>pointMoved(unsigned)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointMoved(unsigned)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_cpT</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_cpC</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_cpB</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_cpX</sender>
   <signal>editingFinished()</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_cpY</sender>
   <signal>editingFinished()</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_cpType</sender>
   <signal>activated(int)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>controlPointEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_N</sender>
   <signal>clicked()</signal>
   <receiver>graph</receiver>
   <slot>nextPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>168</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_P</sender>
   <signal>clicked()</signal>
   <receiver>graph</receiver>
   <slot>prevPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_editx</sender>
   <signal>toggled(bool)</signal>
   <receiver>graph</receiver>
   <slot>enableEditX(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_edity</sender>
   <signal>toggled(bool)</signal>
   <receiver>graph</receiver>
   <slot>enableEditY(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_New</sender>
   <signal>clicked()</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>addControlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph</sender>
   <signal>addControlPoint(value,value)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>addControlPoint(value,value)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph</sender>
   <signal>moveControlPoint(value,value)</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>moveCurrentControlPoint(value,value)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph</sender>
   <signal>deleteControlPoint()</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>deleteCurrentControlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Delete</sender>
   <signal>clicked()</signal>
   <receiver>Envelope_Dlg</receiver>
   <slot>deleteCurrentControlPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>155</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateControlPointUI(unsigned)</slot>
  <slot>controlPointMoved(unsigned)</slot>
  <slot>controlPointEdited()</slot>
  <slot>addControlPoint()</slot>
  <slot>addControlPoint(value,value)</slot>
  <slot>moveCurrentControlPoint(value,value)</slot>
  <slot>deleteCurrentControlPoint()</slot>
 </slots>
</ui>
